### ISSUE TITLE
Bots do not start when trying to run via systemd.

### ISSUE THREAD

**Issue by BunnyApocalypse** [slack-ruby/slack-ruby-client#213]
I have made a bot called [dashbot](https://github.com/BunnyApocalypse/dashbot), and that is how I've mostly been testing this issue, but it also appears an issue with your realtime hello example (the only other thing I've tried to test this with).

Basically, when I try to start the slack-ruby-client via a systemd service, the service says that it has successfully started however the client does not seem to launch until I stop the systemd service, at which point it finishes starting and then immediately is stopped.

You can view the service file in dashbot if you want to see how it is called but here is the output:

Upon starting (it then hangs like this):
```
● dashbot.service - Runs dashbot
   Loaded: loaded (/usr/lib/systemd/system/dashbot.service; disabled; vendor preset: disabled)
   Active: active (running) since Mon 2018-08-06 15:58:09 CDT; 4s ago
 Main PID: 29476 (ruby-mri)
    Tasks: 3 (limit: 4915)
   Memory: 36.1M
   CGroup: /system.slice/dashbot.service
           └─29476 /usr/bin/ruby-mri /bin/dashbot

Aug 06 15:58:09 localhost.localdomain systemd[1]: Started Runs dashbot.
```
Upon stopping:
```
 dashbot.service - Runs dashbot
   Loaded: loaded (/usr/lib/systemd/system/dashbot.service; disabled; vendor preset: disabled)
   Active: inactive (dead)

Aug 06 15:16:28 localhost.localdomain ruby[26431]: Connection has been disconnected.
Aug 06 15:16:28 localhost.localdomain systemd[1]: Stopped Runs dashbot.
Aug 06 15:58:09 localhost.localdomain systemd[1]: Started Runs dashbot.
Aug 06 15:58:43 localhost.localdomain systemd[1]: Stopping Runs dashbot...
Aug 06 15:58:43 localhost.localdomain ruby[29476]: I, [2018-08-06T15:58:10.344298 #29476]  INFO -- : post https://slack.com/api/rtm.start
Aug 06 15:58:43 localhost.localdomain ruby[29476]: I, [2018-08-06T15:58:10.697421 #29476]  INFO -- Status: 200
Aug 06 15:58:43 localhost.localdomain ruby[29476]: Successfully connected.
Aug 06 15:58:43 localhost.localdomain ruby[29476]: Client is about to disconnect
Aug 06 15:58:43 localhost.localdomain ruby[29476]: Client has disconnected successfully!
Aug 06 15:58:43 localhost.localdomain systemd[1]: Stopped Runs dashbot.
```
When I normally run the bot from my command line, it successfully runs and functions as expected, this issue only exists when run via systemd.

Interestingly, when I change the systemd exec line from `Exec=/bin/ruby /bin/dashbot` to `Exec=/bin/irb /bin/dashbot`, the bot successfully starts and functions as expected, although obviously running the bot through irb is less than optimal.

Feel free to ask me if you need any more info.

**Comment by dblock** [comment]
Since you can see the bot actually try to start on stop of the service it looks like it doesn't actually start any of the celluloid-io actors. I would first try this with faye-websocket instead of celluloid-io to narrow things down, and then maybe try implementing #210.

**Comment by BunnyApocalypse** [comment]
In my bot I require faye/websocket, is that not already specifying to use faye instead of celluloid? I don't think that I have the Ruby chops to help properly implement support for a new async gem, but if I am properly using faye in my code and that seems to be the only way forward I could give it a swing.

**Comment by dblock** [comment]
`faye/websocket` is one library, `celluloid` is the other, since you're already using the first try the second, first?

**Comment by BunnyApocalypse** [comment]
Switching to celluloid did not help the issue.

**Comment by BunnyApocalypse** [comment]
Interestingly though, it does get partially through the startup process and outputs:
I, [2018-08-08T10:29:50.962416 #20421]  INFO -- : Celluloid 0.17.3 is running in BACKPORTED mode. [ http://git.io/vJf3J ]
Before hanging until I tell systemd to stop the service, at which point it does the same as faye did.

**Comment by dblock** [comment]
Weird. Honestly I don't understand what happens under systemd that creates this behavior - @BunnyApocalypse you should do some research and maybe ask on both faye and celluloid mailing lists/forums. Try to find a tiny repro for them too.

**Comment by BunnyApocalypse** [comment]
Wow, thanks to this suggestion I ended up finding out that my bot was not working for a completely unrelated reason... 

The STDOUT.sync = true did end up fixing the hanging issue, although I think that it might have been hanging in terms of it's output but actually running.

The actual reason that my bot was failing was due to the defined? function not working as a user that wasn't me for whatever reason, switching my logic to &. instead ended up fixing it. Thanks for all of the help, and sorry for wasting your time on a problem that wasn't actually related to your gem at all.
